src/software/backend/replay_backend.cpp

```diff
@@ -0,0 +1,43 @@
+#include "replay_backend.h"
+
+#include "software/util/design_patterns/generic_factory.h"
+
+ReplayBackend::ReplayBackend(const std::string& replay_input_dir)
+    : Backend(),
+      replay_reader(replay_input_dir),
```

Comment:
does this fail nicely when the `replay_input_dir` isn't valid?

Reply:
`ReplayReader` will throw